Yoast/YoastSEO.js
# Add worker tests

## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Added specs for AnalysisWebWorker.
* Added spec helpers: `getAssessment`.
* Added a type (string) in the assessors to easily check which type of assessor it is in the tests.
* Changed the cornerstone specs to include checking for the config.
* Changed the location of the spec helpers to the `specHelpers` directory so that whole folder can be ignored by Jest.
* Removed the passing of a non-existing config value in the cornerstone seo assessor.



Created issues for missing failsafes for: 
- `analyze` issue: https://github.com/Yoast/YoastSEO.js/issues/1774
- `analyzeRelatedKeywords` issue: https://github.com/Yoast/YoastSEO.js/issues/1775
- `runResearch` issue: https://github.com/Yoast/YoastSEO.js/issues/1773

## Test instructions

This PR can be tested by following these steps:

* Read the worker test files and check if everything that you would expect is actually tested.
* Run `yarn test`.

Fixes #1740
